### ISSUE TITLE
run: ichigo 0.4

### ISSUE THREAD

**Issue by tikikun** [homebrewltd/ichigo#102]
## Goal
Incrementally improve the main qualities

## Methodology
- Fuse the phase 3 and phase 2 together
- Add long random noise to reduce task misalignment and better overall generalization

**Comment by bachvudinh** [comment]
## Goal
- Improve noise handle ability of model
- Improve multiturn ability
## Methodology
- Describe above

## Hyperparams
Parameter | Value
--- | ---
Epoch | 1
Global batch size | 256
Learning Rate | 7e-5
Learning Scheduler | LambdaLR with warmup
Optimizer | AdamW Fused
Warmup Steps | 80
Weight Decay | 0.005
Gradient Checkpointing | Full
Max length | 4096
Precision | bf16

## Results
- MMLU:
  + Checkpoint 7000:
  
![Screenshot from 2024-11-08 09-54-26](https://github.com/user-attachments/assets/cb849619-93da-4a2c-a3ed-88794f65df36)

  + Checkpoint 8000:
  
![Screenshot from 2024-11-08 09-54-45](https://github.com/user-attachments/assets/0855e790-7d92-4a86-9d42-ddd2f338c8e2)
  
  + Checkpoint end epoch:
   
  
  

- Audio-Bench:
  + Speech Instruction (GPT-4 judge):
    ** Checkpoint 7000:
    Alpaca : 3.54
    Open-hermes: 3.51

     ** Checkpoint 8000:
    Alpaca : 3.53
    Open-hermes: 3.51

    ** Checkpoint end epoch:
    Alpaca : 3.52
    Open-hermes: 3.5
  
  + ASR (WER score):
    ** Checkpoint end epoch:
    Score~ 0.22

## Learnings
- Model return repetition response when processing the low quality audio

## Quicklinks
- Data: 
  + Training data: 

- Checkpoints: 
  + Checkpoint step 7000: https://huggingface.co/jan-hq/llama3-s-instruct-v0.3-checkpoint-7000.
  + Checkpoint end epoch: https://huggingface.co/jan-hq/llama3-s-instruct-v0.3-checkpoint-last.
